src/NTP/Util.hs

```diff
@@ -64,3 +69,14 @@ selectIPv6 = find (\a -> addrFamily a == AF_INET6)
 
 selectIPv4 :: [AddrInfo] -> Maybe AddrInfo
 selectIPv4 = find (\a -> addrFamily a == AF_INET)
+
+withSocketsDoLifted :: (MonadIO m, MonadMask m) => m a -> m a
+withSocketsDoLifted action = do
+    -- `MonadBaseControl IO` is not provided sometimes, so hack with threads
+    -- is used
+    bracket (liftIO $ newTVarIO False)
+            (liftIO . atomically . flip writeTVar True) $
+            \exited -> do
+                liftIO . void . forkIO $
```

Comment:
`withSocketsDo` has signature `IO a -> IO a`, as argument I have to provide an action, which works till I need networking capabilities.
This it the only way I see, without using `liftBaseWith`